### ISSUE TITLE
ecma6 not supported

### ISSUE THREAD

**Issue by nattlip** [megaboich/js-map-parser#6]
i get an error in vs 2015 with the map parser with  "=>"  in my code in ecma6 javascript

**Comment by megaboich** [comment]
Yes, I admit. Currently extension is using https://github.com/sebastienros/jint which is EcmaScript5.1 compatible. There is a promising project https://github.com/sebastienros/esprima-dotnet which supports ES2016. But still some efforts are required to switch.

**Comment by nattlip** [comment]
Outlook voor Android<https://aka.ms/ghei36> downloaden



On Wed, Dec 7, 2016 at 9:43 AM +0100, "megaboich" <notifications@github.com<mailto:notifications@github.com>> wrote:


Yes, I admit. Currently extension is using https://github.com/sebastienros/jint which is EcmaScript5.1 compatible. There is a promising project https://github.com/sebastienros/esprima-dotnet which supports ES2016. But still some efforts are required to switch.

**Comment by msamaann** [comment]
would be really nice if this plugin supported ecma6

**Comment by robert777robert777** [comment]
[JINT 3.X (beta)](https://www.nuget.org/packages/Jint/3.0.0-beta-2036) DOES support ES6 and even some features up to ECMAScript 2020.
Any chance to build with JINT 3.X? According to jint docs it's supposed to be compatible

**Comment by chcg** [comment]
This plugin uses https://github.com/sebastienros/jint/tree/release/2.x/Jint/Parser at https://github.com/megaboich/js-map-parser/tree/master/source/Jint.Parser. Jint 3.X has no longer a own parser, but uses https://github.com/sebastienros/esprima-dotnet/tree/main/src/Esprima, see e.g. https://github.com/sebastienros/jint/blob/main/Jint.Tests/Parser/JavascriptParserTests.cs.
I'm afraid that might be no dropin replacement but needs some major adaptions.

**Comment by mlisowsk** [comment]
Well at least you can put comments around problematic code like import statements:
// jsparser:off
import * from "abc"
// jsparser:on
and the JS Map Parser will cooperate again
